packages/discovery-provider/src/tasks/entity_manager/entities/playlist.py

```diff
@@ -346,8 +346,9 @@ def process_playlist_data_event(
         # Update the playlist_record when the corresponding field exists
         # in playlist_metadata
         if key == "playlist_contents":
-            if not playlist_metadata.get(key) or playlist_record.is_album:
+            if not playlist_metadata.get(key):
                 continue
+            # TODO: don't allow adding non-owned tracks to albums
```

Comment:
this seems important to get in before merging, right?

Reply:
Added this and also integration tests to cover the behavior. Please take another look when you have a chance @isaacsolo! Thanks so much for your help 🙏